src/Typing.js

```diff
@@ -16,11 +16,13 @@ class Typing extends Component {
   };
 
   componentWillReceiveProps({ children }) {
-    if (
-      children !== undefined &&
-      JSON.stringify(children) !== JSON.stringify(this.props.children)
-    ) {
-      this.resetState();
+    if (children !== undefined) {
+      var oldKeys = this.props.children.map( child => child.key);
```

Comment:
Changed var -> const, although this solution doesn't check the entire tree for changes, just the first level. And even then, if the key stays the same, this won't even work one level deep.